util/chplenv/chpl_3p_libfabric_configs.py

```diff
@@ -0,0 +1,24 @@
+import chpl_locale_model, third_party_utils
+from utils import error, memoize
+
+
+@memoize
+def get_uniq_cfg_path():
+    return third_party_utils.default_uniq_cfg_path()
+
+@memoize
+def get_link_args(libfabric):
+    libs = []
+    if libfabric == 'system':
+      # Try using pkg-config to get the libraries to link
+      # libfabric with.
+      pclibs = third_party_utils.pkgconfig_get_link_args(
+                       'libfabric', system=True)
+      for pcl in pclibs:
+        libs.append(pcl)
+        if pcl.startswith('-L'):
+          libs.append(pcl.replace('-L', '-Wl,-rpath,', 1))
```

Comment:
Can you add a comment to this? Is using rpath a side effect of the above call using `static=True`?

Reply:
Scratch the original question. For other 3p things (say jemalloc) we have the rpath stuff in runtime/etc/Makefile.jemalloc-jemalloc. What's different here that we have this logic here?

Reply:
I'll look into putting this in a `runtime/etc/Makefile.libfabric-system`.

Reply:
Certainly this is doable.  What is the the better place to add the `-Wl,-rpath` link-time option, though?  I think the util/chplenv scripts are actually the better place.  We could get rid of the `<pkg>_LIB_DIR` settings for `pkg=GMP,HWLOC,JEMALLOC,RE2,QTHREAD` entirely if we moved generating the `-Wl,-rpath` options into `third_party_utils.pkgconfig_get_link_args`.  It seems preferable to put it in that one single place.

Reply:
So long as we're consistent, I don't think I have a strong preference for the location.

If you think chplenv is the right spot, let's go with that (and either make that change to other 3p packages in this PR or open an issue for changing them.)

Reply:
https://github.com/Cray/chapel-private/issues/580 records the need/desire for this.